lesson_9/lesson_9_authentication.py

```diff
@@ -0,0 +1,271 @@
+                break
+            print('What is your username?')
+            name = ask_name()
+            user = find_user_by_name(name)
+            if not user:
+                print('Error. There is no such user. Try again.')
+                continue
+            while True:
+                print('You need to enter your password in order to change your name.')
+                password = ask_password()
+                if user.password == password:
+                    while True:
+                        new_name = input('Set your new username: ')
+                        if new_name == user.name:
+                            print('Error. New name must differ from previous. Try again.')
+                            continue
+                        if find_user_by_name(new_name):
+                            print('Error. This username is already taken.')
+                            continue
+                        if not check_name(new_name):
+                            continue
+                        else:
+                            new_user = User(new_name, user.password, user.age, user.key)
+                            new_user.save(user)
+                            print('Successfully changed your name.')
+                            break
+                else:
+                    print('Incorrect password. Try again')
+                    continue
+                break
+            break
+
+    @staticmethod
+    def change_age():
+        check_logins = turn_logins_into_class_user()
+        while True:
+            if not check_logins:
+                print('There are no registered users. You should registrate.')
+                break
+            print('What is your username?')
+            name = ask_name()
+            user = find_user_by_name(name)
+            if not user:
+                print('Error. There is no such user. Try again.')
+                continue
+            while True:
+                print('You need to enter your password in order to change your age.')
+                password = ask_password()
+                if user.password == password:
+                    while True:
+                        new_age = ask_n_check_age()
+                        if new_age == user.age:
+                            print('Error. New age must differ from previous. Try again.')
+                            continue
+                        else:
+                            new_user = User(user.name, user.password, new_age, user.key)
+                            new_user.save(user)
+                            print('Successfully changed your age.')
+                            break
+                else:
+                    print('Incorrect password. Try again')
+                    continue
+                break
+            break
+
+    @staticmethod
+    def change_password():
+        check_logins = turn_logins_into_class_user()
```

Comment:
also we use this function here and after that we lauch `find_user_by_name` that execute `turn_logins_into_class_user` function again.. better to avoid it if we can 😎